### DISCUSSION TITLE
Audio output?

### DISCUSSION THREAD

**Discussion by thefinn93** [gokrazy/gokrazy#238] [General]
I'm interested in using gokrazy to make audio come out of my raspberry pi. Is that something anyone has done before? I'm not seeing anything in the documentation or anywhere else online. My goal is to make a simple internet radio playback device.

**Comment by damdo** [comment]
Hey @thefinn93, that sounds like an interesting project!
Anisse worked on something vaguely similar (playback of audio on gokrazy) called beatbox.
You can see it here: https://github.com/anisse/beatbox

**Reply by thefinn93** [reply]
awesome, this is exactly the kind of starting point I was looking for! Based on a little bit of poking around, it uses a custom kernel with [some magic to enable audio support](https://github.com/anisse/beatbox-kernel/commit/f597f35750fa1703a81bef621056225faaec3237), then a [go library to interface with it](https://github.com/anisse/alsa).

**Reply by thefinn93** [reply]
following up on this: I'd like to avoid using a custom kernel if possible, and it seems like I can at least do the `config.txt` update if I add this to `config.json`:
```json
    "BootloaderExtraLines": ["dtparam=audio=on"]
```

however, I'm not seeing a way to tweak the `configAddendum` which seems to be the other part of beatbox's custom kernel modifications. Is that possible? or will I also have to maintain a custom kernel

**Reply by stapelberg** [reply]
There is no way to change the kernel configuration at instance-build-time (this would involve building the kernel on-demand).

You can however suggest the needed changes as pull request for https://github.com/gokrazy/kernel, and as long as they don’t break anything else, we can merge it.

**Reply by thefinn93** [reply]
That sounds great! I have [opened a PR](https://github.com/gokrazy/kernel/pull/554).

Just noting for people interested in this in the future: Anisse's alsa library hard-codes the audio output device to one that didn't work for me (I'm guessing it's the HDMI audio output). I had to fork the library to allow specifying the output device, and it worked. I'll try to come back here and post my final results when I have something reasonably workable.
